### ISSUE TITLE
[Question] How to test watcherSaga with all effects?

### ISSUE THREAD

**Issue by FourwingsY** [redux-saga/redux-saga#1318]
I am using this type of sagas, and want to test:
"is watcherSaga takes action and call someSaga correctly?"
```javascript
export function* watcherSaga() {
  yield all([
    takeLatest(action, someSaga),
    takeEvery(otherAction, otherSaga),
  ])
}
```

and in my test code,
```javascript
describe('watcherSaga', () => {
  const watchAll = watcherSaga()
  it('should watch otherAction', () => {
    const allDescriptor = watchAll.next().value
    const effects = allDescriptor['ALL'] // <- this is awkward
    expect(effects).toContainEqual(takeEvery(otherAction, otherSaga))
  })
})
```

is there any better way to fix `['ALL']` with testing all effects?

**Comment by Andarist** [comment]
This should be possible
```js
describe('watcherSaga', () => {
  const watchAll = watcherSaga()
  it('should watch otherAction', () => {
    const effect = watchAll.next().value
    expect(effect).toEqual(all([
        takeLatest(action, someSaga),
        takeEvery(otherAction, otherSaga)
    ])),
  })
})
```

**Comment by FourwingsY** [comment]
Hmm, testing with whole effects?
I wanted to test each actions called... How do you think, this approach is not necessary?

**Comment by Andarist** [comment]
Sorry, ain't sure what do you mean - could you rephrase?

**Comment by FourwingsY** [comment]
actually, my test code is grouped by worker sagas

```javascript
describe('someSaga', () => {
  const someGenerator = someSaga()
  it('test someSaga', () => {...})
  // ...
  it('should be forked by watcherSaga', () => {
    const watchAll = watcherSaga()
    const allDescriptor = watchAll.next().value
    const effects = allDescriptor['ALL']
    expect(effects).toContainEqual(takeEvery(action, someSaga))
  })
})
```

do you think this fork test could be replaced by testing watcher saga?

**Comment by Andarist** [comment]
I recommend you to read - http://blog.scottlogic.com/2018/01/16/evaluating-redux-saga-test-libraries.html .

If you want to test by iterating generators, then I'd go with the snippet I've written above - but in general it might be better to use approaches mentioned in the article.

**Comment by neurosnap** [comment]
Closing due to inactivity.  If this is still an issue then please reply and I will reopen and try to help.  Thanks!
